### PR TITLE
feat(ListFormat): add `ListFormat` component

### PR DESCRIPTION
**Pull request by langz**
[PR Preview](https://eufemia-git-feat-format-list-eufemia.vercel.app/uilib/components/list-format/).

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+```
+
+## Description
+
+A ready-to-use list formatter. Use it wherever you have to display a list of strings, numbers, or elements.
+
+Good reasons for why we have this is to:
+
+- uniform the creation and formatting of lists.
+- Any other good reasons?
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:20]:
```suggestion
- Any other good reasons?
```

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    | 'A'
+    | 'i'
+    | 'I'
+    | '1'
+    | 'circle'
+    | 'disc'
+    | 'square'
+    | undefined
+
+  value?: any
```
Comment (langz) [inline packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:37]:
```suggestion
  value?: any
```
TODO

File: packages/dnb-eufemia/src/components/list-format/ListFormatDocs.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { PropertiesTableProps } from '../../shared/types'
+
+export const ListFormatProperties: PropertiesTableProps = {
+  value: {
+    doc: 'value to format',
+    type: 'any',
```
Comment (langz) [inline packages/dnb-eufemia/src/components/list-format/ListFormatDocs.ts:6]:
```suggestion
    type: 'any',
```
TODO

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+## Import
+
+```tsx
+import { ListFormat } from '@dnb/eufemia'
+```
+
+## Description
+
+A ready-to-use list formatter. Use it wherever you have to display a list of strings, numbers, or elements.
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:15]:
```suggestion
A ready-to-use list formatter. Use it wherever you have to display a list of strings, numbers, or React components (JSX).
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 1 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:20]: - Supports translation and localization.
- Built on top of web standards.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+showTabs: true
+---
+
+import { ListFormat } from '@dnb/eufemia/src'
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:5]:
```suggestion
```

Do we need this import?

**Review by tujoworker — COMMENTED** [review]
Nice work so far 🤩

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 1 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:20]: Thanks, I added this now :)

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    localProps,
+    defaultProps,
+    ListFormat
+  )
+  const {
+    value,
+    format,
+    variant = 'text',
+    listType,
+    children, //eslint-disable-line
```
Comment (langz) [inline packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:67]: I've not added support for handling children as React.ReactNode yet.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 5 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:5]: No no, thanks 🙇

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx (hunk 6 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx]: Added

**Comment by langz** [comment]
> Nice work so far 🤩

Thanks for the help and a round of review @tujoworker :)
I'm ready for a new round of review now, as I think it looks pretty good now.
Here's a link to the docs https://eufemia-git-feat-format-list-eufemia.vercel.app/uilib/components/list-format/

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+## Import
+
+```tsx
+import { ListFormat } from '@dnb/eufemia'
+```
+
+## Description
+
+A ready-to-use list formatter. Use it wherever you have to display a list of React components / JSX which includes strings, numbers, etc.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:13]:
```suggestion
A ready-to-use list formatter. Use it wherever you have to display a list of React components (JSX) which includes strings, numbers, etc.
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+## Import
+
+```tsx
+import { ListFormat } from '@dnb/eufemia'
+```
+
+## Description
+
+A ready-to-use list formatter. Use it wherever you have to display a list of React components (JSX) which includes strings, numbers, etc.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:13]:
```suggestion
A ready-to-use list formatter. Use it wherever you have to display a list of strings, numbers, or React components (JSX).
```

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+    const ListElement = variant.startsWith('ol') ? Ol : Ul
+
+    return <ListElement type={listType}>{list}</ListElement>
+  }, [format, list, locale, variant, listType])
+
+  return result
+}
+
+// Support for "ListFormat.format(list)" for non-React usage
+ListFormat.format = listFormat
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:92]:
```suggestion
```

Depends on how we want devs to use the interceptor function to be used.

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+
+export function listFormat(
+  list: Array<React.ReactNode> | React.ReactNode,
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:95]:
```suggestion
  list: Array<React.ReactNode>,
```

I think it always should be an array.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+      {listFormat(
+        [
+          <React.Fragment key="a">A</React.Fragment>,
+          <>
+            <b>B</b>
+          </>,
+          <>C</>,
+          'D',
+          123,
+          <a
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:27]: I suggest that we rather use our Anchor element/component directly.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+      <ListFormat
+        value={[
+          <React.Fragment key="a">A</React.Fragment>,
+          <>
+            <b>B</b>
+          </>,
+          <>C</>,
+          'D',
+          123,
+          <a
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:61]: Same here.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+        <ListFormat
+          value={[
+            <React.Fragment key="a">A</React.Fragment>,
+            <>
+              <b>B</b>
+            </>,
+            <>C</>,
+            'D',
+            123,
+            <a
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:92]: And here.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+
+export const Inline = () => {
+  return (
+    <ComponentBox data-visual-test="list-format-inline">
+      <P>
+        This is before the component{' '}
+        <ListFormat
+          value={[
+            123,
+            <a
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:120]: And here.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+Good reasons for why we have this is to:
+
+- uniform the creation and formatting of lists.
+- Supports translation and localization.
+- Built on top of web standards.
+
+The component is designed to maximum display 10-20 items.
+If you need to display more items than that, consider a different design, and perhaps using [Pagination](/uilib/components/pagination) and/or [InfinityScroller](/uilib/components/pagination/infinity-scroller)
+
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:23]:
```suggestion
When the `variant` property is set to a non-`text` variant, it uses [Lists](/uilib/elements/lists/) to render the given list.

```

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx [hunk 16]
```diff
@@ -0,0 +1,154 @@
...
+
+  const list = useMemo(() => {
+    const isListVariant = variant !== 'text'
+    if (children) {
+      return isListVariant
+        ? React.Children.map(children, (child: React.ReactNode, index) => {
+            return <Li key={index}>{child}</Li>
+          })
+        : children
+    }
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:66]:
```suggestion
const list = value || children
```

I think we should handle `value` and `children` the same. So just create a new

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx [hunk 17]
```diff
@@ -0,0 +1,154 @@
...
+    return isListVariant
+      ? value?.map((value, index) => (
+          <Li key={index}>
+            {React.isValidElement(value)
+              ? value
+              : convertJsxToString(value)}
+          </Li>
+        ))
+      : value
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:75]:
```suggestion
      ? list?.map((value, index) => (
          <Li key={index}>
            {React.isValidElement(value)
              ? value
              : convertJsxToString(value)}
          </Li>
        ))
      : list
```

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx [hunk 18]
```diff
@@ -0,0 +1,154 @@
...
+            return <Li key={index}>{child}</Li>
+          })
+        : children
+    }
+    return isListVariant
+      ? value?.map((value, index) => (
+          <Li key={index}>
+            {React.isValidElement(value)
+              ? value
+              : convertJsxToString(value)}
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:72]: Do we have good reasons to use `convertJsxToString` to render the list child to a string? I can't come up with a good reason actually. 

@joakbjerk do you have any input on this?

File: packages/dnb-eufemia/src/components/list-format/ListFormatDocs.ts [hunk 19]
```diff
@@ -0,0 +1,39 @@
+import { PropertiesTableProps } from '../../shared/types'
+
+export const ListFormatProperties: PropertiesTableProps = {
+  value: {
+    doc: 'The value to format.',
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/list-format/ListFormatDocs.ts:5]:
```suggestion
    doc: 'The value to format. Can be given as `children` instead.',
```

File: packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx [hunk 20]
```diff
@@ -0,0 +1,590 @@
...
+            key="github"
+          >
+            Link to Eufemia's Github Repo
+          </a>,
+        ]}
+      />
+    )
+
+    expect(container).toHaveTextContent(
+      "A, B, C, D, 123 og Link to Eufemia's Github Repo"
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx:53]: Having a "so long" text like the anchor provides, we miss a bit of the important thing, testing the "og". So maybe just one work is good enough?

File: packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx [hunk 21]
```diff
@@ -0,0 +1,590 @@
...
+})
+
+describe('ListFormat aria', () => {
+  it('should validate', async () => {
+    const Component = render(<ListFormat value={[1]} />)
+    expect(await axeComponent(Component)).toHaveNoViolations()
+  })
+})
+
+describe('listFormat', () => {
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx:479]: Could you add a couple of tests that test edge cases? Like what if a child has `undefined`, `false`, `0`, `null`, `function`, or an `object` as its value?

File: packages/dnb-eufemia/src/extensions/forms/Value/ArraySelection/ArraySelection.tsx [hunk 22]
```diff
@@ -62,69 +34,33 @@ function ArraySelection(props: Props) {
...
-        data?.map?.(({ title }, index) =>
-          isListVariant ? (
-            <Li key={index}>{convertJsxToString(title)}</Li>
-          ) : (
-            convertJsxToString(title)
-          )
-        ) || value
-      )
+      valueToUse =
+        data?.map?.(({ title }) => convertJsxToString(title)) || value
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/extensions/forms/Value/ArraySelection/ArraySelection.tsx:38]: Because we have `convertJsxToString`, I don't find any further arguments to do the same in the ListFormat component.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+## Formatting only
+
+You can use the `listFormat` function without using the React Component `ListFormat`.
+
+```ts
+import { listFormat } from '@dnb/eufemia/components/NumberFormat'
```
[the remark this answers is not in the document]
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:29]: I don't have any particular opinion on this.
I feel like we usually do `import { listFormat } from '@dnb/eufemia/components/ListFormat'`, can't remember seeing that we often do `ComponenName.function`.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx (hunk 11 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:27]: Reason for using `<a>` is because I wanted to see if it works for the Value.Upload component I'm making, and tthere I just copied from UploadFileListCell which uses `<a>`

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx (hunk 12 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:61]: Reason for using `<a>` is because I wanted to see if it works for the Value.Upload component I'm making, and tthere I just copied from UploadFileListCell which uses `<a>`, but we can sure change in this example 👍

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx (hunk 14 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:120]: Reason for using `<a>` is because I wanted to see if it works for the Value.Upload component I'm making, and tthere I just copied from UploadFileListCell which uses `<a>`, but we can sure change in this example 👍

File: packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx (hunk 20 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx:53]: Improved

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx (hunk 10 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx]: I've tried to make the change now.
But not 100% sure  how I'll make it work, as I think we would have to support React.ReactNode, or else we'll have to do some more manipulation before passing children to this function

**Comment by langz** [comment]
Thanks for the review, @tujoworker 
I've addressed most of the comments for now, and will not look more into it until tomorrow.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx [hunk 24]
```diff
@@ -0,0 +1,192 @@
...
+          <>
+            <b>B</b>
+          </>,
+          <>C</>,
+          'D',
+          123,
+          <Anchor
+            target="_blank"
+            href="https://github.com/dnbexperience/eufemia"
+            className="dnb-anchor"
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:30]:
```suggestion
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx [hunk 25]
```diff
@@ -0,0 +1,192 @@
...
+          <>
+            <b>B</b>
+          </>,
+          <>C</>,
+          'D',
+          123,
+          <a
+            target="_blank"
+            href="https://github.com/dnbexperience/eufemia"
+            className="dnb-anchor"
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:64]:
```suggestion
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx [hunk 26]
```diff
@@ -0,0 +1,192 @@
...
+            <>
+              <b>B</b>
+            </>,
+            <>C</>,
+            'D',
+            123,
+            <Anchor
+              target="_blank"
+              href="https://github.com/dnbexperience/eufemia"
+              className="dnb-anchor"
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:95]:
```suggestion
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx [hunk 27]
```diff
@@ -0,0 +1,192 @@
...
+    <ComponentBox data-visual-test="list-format-inline">
+      <P>
+        This is before the component{' '}
+        <ListFormat
+          value={[
+            123,
+            <Anchor
+              target="_blank"
+              href="https://github.com/dnbexperience/eufemia"
+              className="dnb-anchor"
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/Examples.tsx:123]:
```suggestion
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/demos.mdx [hunk 28]
```diff
@@ -0,0 +1,31 @@
+---
+showTabs: true
+---
+
+import * as Examples from './Examples'
+
+## Demos
+
+### Value
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/demos.mdx:9]:
```suggestion
### Basic usage
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 29]
```diff
@@ -0,0 +1,40 @@
...
+import { ListFormat } from '@dnb/eufemia'
+```
+
+## Description
+
+A ready-to-use list formatter. Use it wherever you have to display a list of strings, numbers, or React components (JSX).
+
+Good reasons for why we have this is to:
+
+- uniform the creation and formatting of lists.
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:17]:
```suggestion
- Uniform the creation and formatting of lists.
```

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx (hunk 18 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx]: I've removed this now, seems to work as expected.

File: packages/dnb-eufemia/src/extensions/forms/Value/ArraySelection/ArraySelection.tsx (hunk 22 above)
Comment (langz) [reply packages/dnb-eufemia/src/extensions/forms/Value/ArraySelection/ArraySelection.tsx:38]: I removed it from ListFormat now.

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx (hunk 10 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx]: Resolved it now, when recieving a single React.ReactNode, I jsut treat it as [React.ReactNode]

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx [hunk 30]
```diff
@@ -0,0 +1,158 @@
...
+          </Li>
+        ))
+      : value
+  }, [value, children, variant])
+
+  const result = useMemo(() => {
+    if (variant === 'text') {
+      return listFormat(list as Array<React.ReactNode>, { locale, format })
```
[the remark this answers is not in the document]
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx]: Resolved :)

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx (hunk 17 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx]: Simplified now :)

**Comment by langz** [comment]
Thanks again for the comments, I've now made a few changes again, addressing the feedback, and simplifying the code.

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx (hunk 16 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx]: Done :)

File: packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx [hunk 31]
```diff
@@ -0,0 +1,654 @@
...
+      ', 0 og 0'
+    )
+  })
+
+  it('accepts array with string values', () => {
+    expect(listFormat(['Foo', 'Bar', 'Baz'])).toBe('Foo, Bar og Baz')
+  })
+
+  it('accepts array with empty string values', () => {
+    expect(listFormat(['', '', ''])).toBe(',  og ')
```
Comment (langz) [inline packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx:541]:
```suggestion
    expect(listFormat(['', '', ''])).toBe('')
```
Should this result in an empty string or  '`,  og '` ? 🤔 
What do we think?

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 23 above)
[the remark this answers is not in the document]
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx]: Yes. The first one does always work. But I think, if we will make the import "easier" accessible, we may go for `ListFormat.format()` – and also in regards, that there will not come other functions.

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx (hunk 9 above)
Comment (tujoworker) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:93]: I think we should go for `ListFormat.format` in this case – to simplify the usage.

File: packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx (hunk 31 above)
Comment (tujoworker) [reply packages/dnb-eufemia/src/components/list-format/__tests__/ListFormat.test.tsx:541]: I don't think so. This edge case needs to be handled by the application. Also because this is what we get from the browser API. 

An app can do this `list.filter(Boolean)` if there is the potential that a value will be falsy.

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 23 above)
[the remark this answers is not in the document]
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx]: Fine by me 🤝

File: packages/dnb-eufemia/src/components/list-format/ListFormat.tsx (hunk 9 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/list-format/ListFormat.tsx:93]: Fine by me 🤝

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 32]
```diff
@@ -0,0 +1,40 @@
...
+
+Good reasons for why we have this is to:
+
+- Uniform the creation and formatting of lists.
+- Supports translation and localization.
+- Built on top of web standards.
+
+The component is designed to maximum display 10-20 items.
+If you need to display more items than that, consider a different design, and perhaps using [Pagination](/uilib/components/pagination) and/or [InfinityScroller](/uilib/components/pagination/infinity-scroller)
+When the `variant` property is set to a non-`text` variant, it uses [Lists](/uilib/elements/lists/) to render the given list.
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:23]:
```suggestion
If you need to display more items than that, consider a different design, and perhaps using [Pagination](/uilib/components/pagination) and/or [InfinityScroller](/uilib/components/pagination/infinity-scroller).

When the `variant` property is set to `text` (default), the browser API [Intl.ListFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat) will be used with additional React components (JSX) support.

When the `variant` is set to a non-`text` variant, it uses [Lists](/uilib/elements/lists/) to render the given list.
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+
+return listFormat(myList, {
+  format: { type: 'disjunction' },
+  locale: 'en-US',
+})
+```
+
+See the following [demo](/uilib/components/list-format/demos/#using-listformat-function) for a more detailed example.
+
+The `listFormat` function supports an object with `{ format, locale }` as the second parameter. `format` and `locale` will accept the same values as documented in [properties](/uilib/components/list-format/properties/) of the `ListFormat` component.
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:43]:
```suggestion
The `formatList` function supports an object with `{ format, locale }` as the second parameter. `format` and `locale` will accept the same values as documented in [properties](/uilib/components/list-format/properties/) of the `ListFormat` component.
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 34]
```diff
@@ -0,0 +1,43 @@
...
+When the `variant` property is set to `text` (default), the browser API [Intl.ListFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat) will be used with additional React components (JSX) support.
+
+When the `variant` is set to a non-`text` variant, it uses [Lists](/uilib/elements/lists/) to render the given list.
+
+## Formatting only
+
+You can use the `listFormat` function without using the React Component `ListFormat`.
+
+```ts
+import { listFormat } from '@dnb/eufemia/components/ListFormat'
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:33]:
```suggestion
import { ListFormat } from '@dnb/eufemia'
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 35]
```diff
@@ -0,0 +1,43 @@
...
+
+return listFormat(myList, {
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:35]:
```suggestion
return formatList(myList, {
```

Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:35]: Or should we go for only `format`? It's just, that it has a config property that is called `format`. Maybe we should find a better name for that one instead?

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 33 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:43]: We kept the name `listFormat` for now 🙏

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 34 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:33]: We kept the name `listFormat` for now 🙏

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 35 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:35]: We kept the name `listFormat` for now 🙏

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx (hunk 33 above)
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:43]:
```suggestion
The `listFormat` function supports an object with `{ format, locale }` as the second parameter. `format` and `locale` will accept the same values as documented in [properties](/uilib/components/list-format/properties/) of the `ListFormat` component.
The function does not support `variant` and `listType`, as it does not return a list, but rather return a string(or JSX if that's provided).
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 36]
```diff
@@ -0,0 +1,44 @@
...
+The component is designed to maximum display 10-20 items.
+If you need to display more items than that, consider a different design, and perhaps using [Pagination](/uilib/components/pagination) and/or [InfinityScroller](/uilib/components/pagination/infinity-scroller).
+
+When the `variant` property is set to `text` (default), the browser API [Intl.ListFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat) will be used with additional React components (JSX) support.
+
+When the `variant` is set to a non-`text` variant, it uses [Lists](/uilib/elements/lists/) to render the given list.
+
+## Formatting only
+
+You can use the `listFormat` function without using the React Component `ListFormat`.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:30]:
```suggestion
You can use the `listFormat` function without using the React Component `ListFormat`, to format strings, numbers, or React components (JSX) as string(or JSX).
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx [hunk 37]
```diff
@@ -0,0 +1,44 @@
...
+return listFormat(myList, {
+  format: { type: 'disjunction' },
+  locale: 'en-US',
+})
+```
+
+See the following [demo](/uilib/components/list-format/demos/#using-listformat-function) for a more detailed example.
+
+The `listFormat` function supports an object with `{ format, locale }` as the second parameter. `format` and `locale` will accept the same values as documented in [properties](/uilib/components/list-format/properties/) of the `ListFormat` component.
+The function does not support `variant` and `listType`, as it does not return a list, but rather return a string(or JSX if that's provided).
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/list-format/info.mdx:44]:
```suggestion
The function does not support `variant` and `listType`, as it does not return a list, but rather return a string.
```

**Review by tujoworker — APPROVED** [review]
Looks good. Thank you for the good work 👏
